src/main/java/net/tirasa/connid/bundles/azure/AzureConnector.java

```diff
@@ -1018,6 +1018,15 @@ private ConnectorObject fromGroup(final Group group, final Set<String> attribute
             builder.addAttribute(AttributeBuilder.build(AzureAttributes.GROUP_ID, group.id));
         }
 
+        if (attributesToGet.contains(PredefinedAttributes.GROUPS_NAME)) {
+            List<String> groupNames = new ArrayList<>();
```

Comment:
Any chance to replace these lines with some smarter Java 8 stream ?

Reply:
something like

```java
List<String> groupNames = lient.getAuthenticated().getAllGroupsForGroup(group.id).
   stream().map(Group::mailNickname).
   collect(Collectors.toList());
```

Reply:
Changes applied